Cargo.toml

```diff
@@ -61,7 +61,7 @@ murmur3 = "0.5.2"
 once_cell = "1"
 opendal = "0.45"
 ordered-float = "4.0.0"
-parquet = "50"
+parquet = "51"
```

Comment:
I'm not sure why previously parquet version is not consistent with arrow version. Different versions could cause errors like:

```
    = note: `RecordBatch` and `arrow_array::record_batch::RecordBatch` have similar names, but are actually distinct types                                                                                                             
note: `RecordBatch` is defined in crate `arrow_array`                                                              
   --> /Users/liangchi/.cargo/registry/src/index.crates.io-6f17d22bba15001f/arrow-array-50.0.0/src/record_batch.rs:72:1                                                                                                                
    |                                                                                                              
72  | pub struct RecordBatch {                                                                                     
    | ^^^^^^^^^^^^^^^^^^^^^^                           
note: `arrow_array::record_batch::RecordBatch` is defined in crate `arrow_array`                                   
   --> /Users/liangchi/.cargo/registry/src/index.crates.io-6f17d22bba15001f/arrow-array-51.0.0/src/record_batch.rs:72:1                                                                                                                
    |                                                                                                                                                                                                                                  
72  | pub struct RecordBatch {                                                                                     
    | ^^^^^^^^^^^^^^^^^^^^^^
    = note: perhaps two different versions of crate `arrow_array` are being used?
```

Reply:
Is it possible to change it to `parquet >= 46 <51`? We faced similar problems in icelake. cc @Xuanwo I still recommend to pin the arrow version since arrow doesn't ensure compatibility during major version upgrades.

Reply:
We can temporarily fix the CI by pinning `arrow` and `parquet`. However, since our users rely on various versions of `arrow`, we cannot force them to always use the latest version. Ultimately, we will determine a MSAV through different methods.

Reply:
And commiting `Cargo.lock` is still recommanded to make sure CI works as expected.

Reply:
Cool, let's pin this first to unlock the ci failure.